third_party/blink/renderer/core/layout/layout_object.cc

```diff
@@ -416,6 +416,21 @@ LayoutObject::~LayoutObject() {
   DCHECK(is_destroyed_);
 #endif
   InstanceCounters::DecrementCounter(InstanceCounters::kLayoutObjectCounter);
+
+  // If recording/replaying and in a nondeterministic execution, allow
+  // style_ to leak, since it may otherwise get destroyed in a
+  // non-desterministic fashion and remove itself from font-fallback-maps
+  // that are accessed deterministically.
+  // See https://linear.app/replay/issue/RUN-1758/fontfallbackmap-items-getting-removed-non-deterministically
+  if (recordreplay::IsRecordingOrReplaying("leak-references") &&
+      recordreplay::AreEventsDisallowed())
+  {
+    // Construct a new scoped_refptr into untracked memory, so the
+    // destructor doesn't get run.  This increments the refcount, and
+    // leaks `style_`.
+    char data[sizeof(scoped_refptr<const ComputedStyle>)];
+    new (data) scoped_refptr<const ComputedStyle>(style_);
```

Comment:
Does `style_.release()` not accomplish the same goal?

This seems like it should also work fine, but it takes me a few more seconds to parse the placement new.

Reply:
Looking into it.

Reply:
Actually yeah that's better.  I didn't see that method when I scanned the declaration of the class.  Good find!